flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolBatchSlotRequestTest.java

```diff
@@ -81,8 +67,10 @@ public static void teardownClass() {
 	 */
 	@Test
 	public void testPendingBatchSlotRequestTimeout() throws Exception {
-		try (final SlotPoolImpl slotPool = new SlotPoolBuilder(mainThreadExecutor)
-				.build()) {
+		try (final SlotPoolImpl slotPool = slotPoolBuilder
+			.setBatchSlotTimeout(Time.milliseconds(2L))
+			.build()) {
+
 			final CompletableFuture<PhysicalSlot> slotFuture = SlotPoolUtils.requestNewAllocatedBatchSlot(
 				slotPool,
 				mainThreadExecutor,
```

Comment:
I think direct main thread executor cannot work here?
Would you give it a check?

Reply:
Thanks for your consideration. I'm still working on this.

Reply:
I've undone this modification in the fix-up commit.

Reply:
To avoid thread leak, it's better to use `try-finally`. A ref can be `SlotPoolPendingRequestFailureTest#testPendingSlotRequestTimeout`.

Reply:
> To avoid thread leak, it's better to use `try-finally`. A ref can be `SlotPoolPendingRequestFailureTest#testPendingSlotRequestTimeout`.

Done.